### PR TITLE
Support clearing of dates.

### LINKED ISSUE #2597: Support clearing of dates
On item page, clicking on the garbage can icon clears the date.

image

This was split from #2280

### PR DESCRIPTION
**Pull request by justinlittman**
closes #2597

## Why was this change made? 🤔
UX

## How was this change tested? 🤨

⚡ ⚠ If this change involves consuming from or writing to another service (or shared file system), ***run [integration test create_object_h2_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test manually in [stage|qa] environment, in addition to specs. ⚡

Amy

### CODE REVIEW

File: app/components/works/publication_date_component.html.erb [hunk 1]
```diff
@@ -21,6 +18,11 @@
       <label for="work_published_day" class="visually-hidden">Publication day</label>
       <%= day_field %>
     </div>
+
+    <%= button_tag type: 'button', class: 'btn btn-sm ms-3', aria: { label: "Clear" },
```
Comment (jcoyne) [inline app/components/works/publication_date_component.html.erb:22]: Can this label be more descriptive than "Clear"? "Clear published date" perhaps?

File: app/components/works/dates_component.html.erb [hunk 2]
```diff
@@ -129,13 +150,20 @@
                 { prefix: prefix, field_name: 'created_range(6i)', prompt: 'day' },
                 data: { date_validation_target: 'day', 
                   date_range_target: 'endDay',
+                  date_clear_target: 'day',
                   date_approximate_target: 'day',
                   action: 'date-range#change date-approximate#change' },
                 id: "#{prefix}_created_range_end_day", class: "form-control" %>
           </div>
+          <%= button_tag type: 'button', class: 'btn btn-sm ms-3', aria: { label: "Clear" },
```
Comment (jcoyne) [inline app/components/works/dates_component.html.erb:158]: Can this label be more descriptive than "Clear"? "Clear created date range" perhaps?

File: app/components/works/dates_component.html.erb [hunk 3]
```diff
@@ -22,40 +22,52 @@
...
           <label for="<%= prefix %>_created_day" class="visually-hidden">Created day</label>
           <%= select_day created_day, { prefix: prefix, field_name: 'created(3i)', prompt: 'day'},
                         data: { date_validation_target: 'day',
-                          date_approximate_target: 'day', 
+                          date_approximate_target: 'day',
+                          date_clear_target: 'day',
                           action: 'date-validation#change date-approximate#change' },
                         id: "#{prefix}_created_day", class: "form-control" %>
         </div>
+        <%= button_tag type: 'button', class: 'btn btn-sm ms-3', aria: { label: "Clear" },
```
Comment (jcoyne) [inline app/components/works/dates_component.html.erb:62]: Can this label be more descriptive than "Clear"? "Clear created date" perhaps?

File: app/components/works/dates_component.html.erb (hunk 2 above)
Comment (justinlittman) [reply app/components/works/dates_component.html.erb]: Yes. Done.

File: app/components/works/publication_date_component.html.erb (hunk 1 above)
Comment (justinlittman) [reply app/components/works/publication_date_component.html.erb]: Yes. Done.

**Comment by justinlittman** [comment]
@amyehodge approved.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**